daemon/api.go

```diff
@@ -234,6 +234,9 @@ func loginUser(c *Command, r *http.Request) Response {
 // It requires the state to be locked
 func UserFromRequest(st *state.State, req *http.Request) (*auth.UserState, error) {
 	// extract macaroons data from request
+	if req == nil {
```

Comment:
That feels like an error on the call site. We shouldn't call *FromRequest if we don't have a request.